### PR TITLE
Added useful message for thrown exception

### PR DESCRIPTION
**Pull request by dgeorgi**
Issue: #48772

Added try catch blocks to catch an exception thrown, add a message about the parameter that couldn't be parsed then throw new exception.

### CODE REVIEW

File: src/Interpreters/ReplaceQueryParameterVisitor.cpp [hunk 1]
```diff
@@ -68,8 +68,18 @@ const String & ReplaceQueryParameterVisitor::getParamValue(const String & name)
     auto search = query_parameters.find(name);
     if (search != query_parameters.end())
         return search->second;
-    else
-        throw Exception(ErrorCodes::UNKNOWN_QUERY_PARAMETER, "Substitution {} is not set", backQuote(name));
+    else {
+        try {
```
Comment (rschu1ze) [inline src/Interpreters/ReplaceQueryParameterVisitor.cpp:72]: `try { throw ... } catch` looks a bit silly :-) What about including all information in the exception in the first place and not using `try/catch`? Alternatively, wrap the entire block in `try` such that a single `catch` may add all context.

File: src/Interpreters/ReplaceQueryParameterVisitor.cpp [hunk 2]
```diff
@@ -84,16 +94,26 @@ void ReplaceQueryParameterVisitor::visitQueryParameter(ASTPtr & ast)
...
+        else
+            data_type->getDefaultSerialization()->deserializeTextEscaped(temp_column, read_buffer, format_settings);
+
+        if (!read_buffer.eof())
+            throw Exception(ErrorCodes::BAD_QUERY_PARAMETER,
+                "Value {} cannot be parsed as {} for query parameter '{}'"
+                " because it isn't parsed completely: only {} of {} bytes was parsed: {}",
+                value, type_name, ast_param.name, read_buffer.count(), value.size(), value.substr(0, read_buffer.count()));
+    }
+    catch (const Exception & e) {
```
Comment (rschu1ze) [inline src/Interpreters/ReplaceQueryParameterVisitor.cpp:110]: Perhaps (didn't check closer) function try blocks may help: https://stackoverflow.com/questions/5612486/when-is-a-function-try-block-useful

File: src/Interpreters/ReplaceQueryParameterVisitor.cpp (hunk 1 above)
Comment (dgeorgi) [reply src/Interpreters/ReplaceQueryParameterVisitor.cpp:72]: Thanks for the feedback! Sorry this is my first time working on an open source project (or anything outside of class projects) I'll take a look and clean it up.

**Comment by alexey-milovidov** [comment]
I will rewrite.

**Comment by alexey-milovidov** [comment]
@dgeorgi please check the implementation here: #49061

**Pull request closed without merging**